lib/galaxy/workflow/modules.py

```diff
@@ -167,6 +167,11 @@ def to_cwl(value, hda_references, step):
             return rval
     elif isinstance(value, list):
         return [to_cwl(v, hda_references=hda_references, step=step) for v in value]
+    elif isinstance(value, dict):
+        # Nested tool state, such as conditionals
+        return {k: to_cwl(v, hda_references=hda_references, step=step) for k, v in value.items()}
+    elif is_runtime_value(value):
```

Comment:
Should this be checked before `isinstance(value, dict)`? From the implementation of `is_runtime_value()`, it looks like a runtime value can also be a special dict.

Reply:
Indeed, thank you!